aip/auth/4111.md

```diff
@@ -0,0 +1,83 @@
+--- 
+aip: 
+  id: 4111
+  scope: auth
+  state: approved
+  created: 2018-07-21
+  updated: 2019-07-21
+permalink: /auth/4111
+redirect_from:
+  - /4111
+---
+
+# Self-signed JWT-based Access Token
+
+JWT access token for Google APIs was introduced in 2014 to provide a more
+efficient auth stack for Cloud APIs being accessed via service accounts by
+bypassing the intermediate step of exchanging client assertions for OAuth
+tokens. It becomes a standard authentication method, such that many Cloud APIs
+accept JWT-based access tokens that are locally signed by the service account
+private key.
+
+**Note:** Because this AIP describes guidance and requirements in a
+language-neutral way, it uses generic terminology which may be imprecise or
+inappropriate in certain languages or environments.
+
+## Guidance
+
+This section describes the general guidance of supporting a self-signed
+JWT-based token.
+
+### Default Authentication Method For Service Account Keys
+
+Considering its efficiency (bypassing the exchanging step), [ADC][0] **should**
+use self-signed JWT as the default authentication flow for service account keys.
+
+In other words, [ADC][0] **should** choose the OAuth flow over self-signed JWT
+only if the scope is explicitly provided to the auth library. 
+
+### Expected Behavior
+
+To support the self-signed JWT-based token, the auth libraries **must** follow
+the steps below:
+
+1. Load the service account ID JSON file. Please note that the self-signed
```

Comment:
link to spec for service account JSON file (even if it's not written yet)

Reply:
Done